File: sample/src/main/java/ru/surfstudio/standard/interactor/common/error/NonInstanceClassCreateException.java
package ru.surfstudio.standard.interactor.common.error;

/**
 * Ошибка возникающая при попытке инстанцирования класса где это запрещено. (Например Utils классы)
 */

public class NonInstanceClassCreateException extends RuntimeException {

    public NonInstanceClassCreateException(Class clazz) {
        super(clazz.getName() + " не может иметь инстанс");
    }
}

File: mvp-widget-sample/src/main/java/ru/surfstudio/android/mvpwidget/sample/interactor/common/error/NonInstanceClassCreateException.kt
package ru.surfstudio.android.mvpwidget.sample.interactor.common.error

/**
 * Ошибка возникающая при попытке инстанцирования класса где это запрещено. (Например Utils классы)
 */

class NonInstanceClassCreateException(clazz: Class<*>) : RuntimeException(clazz.name + " не может иметь инстанс")
